source/ConditionSet.cpp

```diff
@@ -92,6 +93,29 @@ bool ConditionSet::IsEmpty() const
 
 
 
+// Do text replacement throughout this ConditionSet.
+ConditionSet ConditionSet::Substitute(const map<string, string> &subs) const
+{
+	ConditionSet result = *this;
+	DataNode someNode = DataNode();
+	for(Expression &expression : result.expressions)
+	{
+		expression.name = Format::Replace(expression.name, subs);
```

Comment:
I get a compile error here, which I fix by adding as line 17 of **ConditionSet.cpp**:  

`#include "Format.h"`